internal/metamorph/response_processor.go

```diff
@@ -1,46 +1,39 @@
 package metamorph
 
 import (
+	"context"
 	"sync"
-	"time"
 
-	"github.com/bitcoin-sv/arc/internal/metamorph/metamorph_api"
 	"github.com/libsv/go-p2p/chaincfg/chainhash"
 )
 
 type StatusResponse struct {
+	ctx      context.Context
 	statusCh chan StatusAndError
-	mu       sync.RWMutex
 
-	Status       metamorph_api.Status
-	Hash         *chainhash.Hash
-	Err          error
-	CompetingTxs []string
+	Hash *chainhash.Hash
 }
 
-func NewStatusResponse(hash *chainhash.Hash, statusChannel chan StatusAndError) *StatusResponse {
+func NewStatusResponse(ctx context.Context, hash *chainhash.Hash, statusChannel chan StatusAndError) *StatusResponse {
 	return &StatusResponse{
+		ctx:      ctx,
 		statusCh: statusChannel,
 		Hash:     hash,
-		Status:   metamorph_api.Status_RECEIVED, // if it got to the point of creating this object, the status is RECEIVED
 	}
 }
 
 func (r *StatusResponse) UpdateStatus(statusAndError StatusAndError) {
-	r.mu.Lock()
-
-	r.Status = statusAndError.Status
-	r.Err = statusAndError.Err
-	r.CompetingTxs = statusAndError.CompetingTxs
-
-	r.mu.Unlock()
-
-	if r.statusCh != nil {
-		r.statusCh <- StatusAndError{
-			Hash:         r.Hash,
-			Status:       statusAndError.Status,
-			Err:          statusAndError.Err,
-			CompetingTxs: statusAndError.CompetingTxs,
+	if r.statusCh != nil && r.ctx != nil {
```

Comment:
nit: I find it usually nice if the happy path is on the left which IMO improves readability.
That principle is nicely explained here: https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88
What do you think?

```
func (r *StatusResponse) UpdateStatus(statusAndError StatusAndError) {
	if r.statusCh == nil || r.ctx == nil {
		return
	}

	select {
	case <-r.ctx.Done():
		return
	default:
		r.statusCh <- StatusAndError{
			Hash:         r.Hash,
			Status:       statusAndError.Status,
			Err:          statusAndError.Err,
			CompetingTxs: statusAndError.CompetingTxs,
		}
	}
}
```

Reply:
Totally agree. Especially in Go this is a good principle. Changed